soc/nxp/imx/imx9/imx95/scmi_cpu_soc.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_NXP_IMX95_SCMI_CPU_SOC_H_
+#define ZEPHYR_NXP_IMX95_SCMI_CPU_SOC_H_
+
+#define CPU_IDX_M33P        0U
```

Comment:
which of these macros are soc-specific and protocol-specific? if there's any protocol-specific stuff (i.e. that won't change across SoCs) then you can just add it to the protocol header. Looking at this header my guess is the CPU indexes are SoC-specific while the CPU modes might be protocol specific?

either way, I'd say there's no need to have this header included in the protocol header. You can just explicitly include it wherever needed?

Reply:
The CPU indexes are SoC-specific. I am not sure if the CPU modes are SoC-specific, I copy them from System Manager which put them into a soc header.

Reply:
Hi Laurentiu,
Thanks for your comments, the struct scmi_cpu_sleep_mode_config is defined in protocol header,
![image](https://github.com/user-attachments/assets/effbb058-f728-433f-a376-0ef7e3634a54)
The macros in **soc.h is the value of the struct, the scmi_cpu protocol depends on the macros, so I think it is better to include the soc.h in protocol header.